Neo4j.Driver/Neo4j.Driver/Internal/Result/ConsumableResultCursor.cs

```diff
@@ -48,19 +52,31 @@ public Task<string[]> KeysAsync()
 
     public Task<IResultSummary> ConsumeAsync()
     {
+        AssertTransactionValid();
         _isConsumed = true;
         return _cursor.ConsumeAsync();
     }
 
+    private void AssertTransactionValid()
+    {
+        PendingError?.EnsureThrown();
+        if (Transaction.IsErrored(out var error))
+        {
+            throw new TransactionTerminatedException(error);
+        }
+    }
+
     public Task<IRecord> PeekAsync()
     {
         AssertNotConsumed();
+        AssertTransactionValid();
```

Comment:
Doing this, each record seems suboptimal, so if we can think of a better place, we should try. similarly for fetch